addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -4614,13 +4601,11 @@ registry.sizing = SnippetOptionWidget.extend({
             moveHandleEl.classList.toggle('d-none', isMobileView);
 
             // Hiding/showing the arrows.
-            if (isGrid) {
+            if (isGrid && !isMobileView) {
                 const moveLeftArrowEl = this.$overlay[0].querySelector('.fa-angle-left');
                 const moveRightArrowEl = this.$overlay[0].querySelector('.fa-angle-right');
-                const showLeft = await this._computeWidgetVisibility('move_left_opt');
-                const showRight = await this._computeWidgetVisibility('move_right_opt');
-                moveLeftArrowEl.classList.toggle('d-none', !showLeft);
-                moveRightArrowEl.classList.toggle('d-none', !showRight);
+                moveLeftArrowEl.classList.add("d-none");
+                moveRightArrowEl.classList.add("d-none");
```

Comment:
Sorry but I don't understand this change. Maybe it was already broken before but before this said: "when in mode grid, the visibility of the left/right arrow depends on `_computeWidgetVisibility` (which checks if mobile)". Now this says "When in grid mode and not in mobile view, the left/right arrow should be hidden"... but it does not seem to say that it should be visible in mobile?

At this point of your PR, this is fully bugged: switching to mobile/desktop in grid mode/col mode is very random.

Sometimes I am getting this:
![image](https://github.com/odoo/odoo/assets/10338094/fb064437-fda9-4c43-b2d2-64947c080c42)
Sometimes this:
![image](https://github.com/odoo/odoo/assets/10338094/e75d26cd-fddd-4d14-b859-b1e14362a3dc)

Just by switching to mobile mode/desktop. Maybe it was the case before your PR I don't know but this code seems weird. If broken before your PR, maybe this requires a stable fix before merging moving code around here :grimacing:

Reply:
> before this said: "when in mode grid, the visibility of the left/right arrow depends on `_computeWidgetVisibility` (which checks if mobile)". Now this says "When in grid mode and not in mobile view, the left/right arrow should be hidden"... but it does not seem to say that it should be visible in mobile?

Hmm, so I was going to say "it's visible by default, this only checks if it should override the visibility", but actually no. This whole condition seems unnecessary if I keep the check in the `_computeWidgetVisibility`.
The visibility depends on the `snippetMove`'s `_computeWidgetVisibility`, which is triggered earlier (through a call to `snippet_option_update` in `_onMobilePreviewClick`).
By the time we arrive here, everything has been done already. This really seems superfluous. I'll remove it.

As for the bug, it also happens from 16.0 on. I'll see how to solve it.